test/libdnf/rpm/solv/test_solv_map.cpp

```diff
@@ -207,6 +207,18 @@ void SolvMapTest::test_iterator_sparse() {
     CPPUNIT_ASSERT_EQUAL((*it2).id, 10);
     CPPUNIT_ASSERT_EQUAL((*it3).id, 6);
 
+    // test move back to begin
+    it2.begin();
+    CPPUNIT_ASSERT_EQUAL((*it2).id, 4);
+
+    // test increment after begin
+    ++it2;
+    CPPUNIT_ASSERT_EQUAL((*it2).id, 6);
+
+    // test end
+    it2.end();
+    CPPUNIT_ASSERT(it2 == map.end());
```

Comment:
What about to add test after end?
```
it2.end();
CPPUNIT_ASSERT(it2 == map.end());
++it2;
CPPUNIT_ASSERT(it2 == map.end());
```

Reply:
To be honest. I added test to increment iterator pointing to the end. But then I removed it. Our implementation works this way. But I don't want to promise it to the user.

In general:
The "end" iterator is used to detect the end of the container (some_iterator == container.end()) and operator++ has undefined behavior for it.
The "end" element usually acts as a placeholder; attempting to access it (dereferencing "end" iterator) results in undefined behavior.

Reply:
Ok, good point.